File: taskrouter/v1/task/update-example-1/update-example-1.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using Twilio;
using Twilio.Rest.Taskrouter.V1.Workspace;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var task = TaskResource.Update(
            assignmentStatus: TaskResource.StatusEnum.Canceled,
            reason: "waiting too long",
            pathWorkspaceSid: "WSXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            pathSid: "WTXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        Console.WriteLine(task.TaskQueueFriendlyName);
    }
}

File: taskrouter/v1/task/update-example-1/update-example-1.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

task = @client.taskrouter
              .workspaces('WSXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
              .tasks('WTXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
              .update(
                 assignment_status: 'canceled',
                 reason: 'waiting too long'
               )

puts task.task_queue_friendly_name
